tests/unit/test_coreneuron.py

```diff
@@ -0,0 +1,191 @@
+    """
+    n = Neurodamus(create_tmp_simulation_config_file)
+    coreneuron_data = Path(SimConfig.coreneuron_datadir)
+    check_directory(coreneuron_data)
+    assert (Path(SimConfig.output_root) / "sim.conf").exists()
+    assert (Path(SimConfig.output_root) / "report.conf").exists()
+    assert (Path(SimConfig.output_root) / "populations_offset.dat").exists()
+
+    n.run()
+
+    assert not coreneuron_data.exists()
+    assert not (Path(SimConfig.output_root) / "sim.conf").exists()
+    assert not (Path(SimConfig.output_root) / "report.conf").exists()
+
+    # check out.dat generated by CoreNEURON
+    assert (Path(SimConfig.output_root) / "out.dat").exists()
+    spikes = np.loadtxt(
+        Path(SimConfig.output_root) / "out.dat", dtype=[("time", "double"), ("gid", "uint32")]
+    )
+    times = spikes["time"]
+    gids = spikes["gid"]
+    npt.assert_allclose(times, ref_timestamps)
+    npt.assert_allclose(gids, ref_gids)
+
+
+@pytest.mark.parametrize(
+    "create_tmp_simulation_config_file",
+    [
+        {
+            "simconfig_fixture": "ringtest_baseconfig",
+            "extra_config": {
+                "target_simulator": "CORENEURON",
+                "inputs": {
+                    "pulse": {
+                        "module": "pulse",
+                        "input_type": "current_clamp",
+                        "delay": 5,
+                        "duration": 50,
+                        "node_set": "RingA",
+                        "represents_physical_electrode": True,
+                        "amp_start": 10,
+                        "width": 1,
+                        "frequency": 50,
+                    }
+                }
+            }
+        }
+    ],
+    indirect=True,
+)
+def test_coreneuron_directmode(create_tmp_simulation_config_file):
+    """Test coreneuron directmode
+    1. empty coreneuron_input folder created and deleted after run
+    2. sim.conf, report.conf, populations_offset.dat are created and deleted after run
+    3. the same spikes data as the file mode (test_coreneuron_filemode)
+    """
+    n = Neurodamus(create_tmp_simulation_config_file, coreneuron_direct_mode=True)
+    coreneuron_data = Path(SimConfig.coreneuron_datadir)
+    assert coreneuron_data.is_dir()
+    assert not any(coreneuron_data.iterdir()), f"{coreneuron_data} should be empty."
+    n.run()
+    assert not coreneuron_data.exists()
+
+    # check the spikes vectors
+    assert not (Path(SimConfig.output_root) / "out.dat").exists()
+    assert n._spike_populations[0][0] == "RingA"
+    assert n._spike_populations[0][1] == 0
+    npt.assert_allclose(n._spike_vecs[0][0].as_numpy(), ref_timestamps)
+    npt.assert_allclose(n._spike_vecs[0][1].as_numpy(), ref_gids)
+
+    assert n._spike_populations[1][0] == "RingB"
+    assert n._spike_populations[1][1] == 1000
+    assert n._spike_vecs[1][0].size() == 0  # no spike on RingB
+    assert n._spike_vecs[1][1].size() == 0
```

Comment:
knitpick: these are gids and timestamps right? Maybe the comment goes on top in this case. Along the line looks like it is only for the first one
```suggestion
    # no spike on RingB
    assert n._spike_vecs[1][0].size() == 0
    assert n._spike_vecs[1][1].size() == 0
```

Reply:
Done